backend/core/parser.py

```diff
@@ -0,0 +1,41 @@
+def parse_credit(text):
+    semester_name = {
+        'U000200001U000300001': '1',
+        'U000200001U000300002': 'S',
+        'U000200002U000300001': '2',
+        'U000200002U000300002': 'W',
+    }
+    lines = text.split('\n')[:-1]
+    columns = []
+    infos = []
+    for i, line in enumerate(lines):
+        line = line.split('\t')
+        if len(columns) != len(line):
+            columns = line
+            continue
+        dic = to_dict(columns, line)
+
+        # 26 : course taking info
+        # 13 : semester info
+        # 9 : total info
+        if len(dic) == 26:
```

Comment:
[core/crawler.py](https://github.com/LastOne817/graduate-adventure/blob/master/backend/core/crawler.py#L57)
```python
rename_list = {
        'schyy': 'year',
        'shtmDetaShtm': 'semester',
        'sbjtNm': 'title',
        'sbjtCd': 'code',
        'ltNo': 'number',
        'acqPnt': 'credit',
        'mrksGrdCd': 'grade',
        'cptnSubmattFgCdNm': 'category',
    }

    new_grade_list = []

    for item in grade_list:
        processed_item = {rename_list[x]: item[x] for x in rename_list.keys()}
        new_grade_list.append(processed_item)
```
I think we can apply a logic like this.
However in this case there are some exceptions like
- type casting to int
- `semester` key